File: backend/app/schemas/audit.py
from datetime import datetime
from decimal import Decimal
from typing import Any

from pydantic import BaseModel


class AuditEvent(BaseModel):
    event_type: str
    title: str
    status: str | None
    timestamp: datetime | None
    # Audit evidence includes nested, persisted JSON such as candidate rankings,
    # policy evidence, and the portfolio constraint snapshot.
    details: dict[str, Any]


class PaymentAuditResponse(BaseModel):
    payment_id: int
    amount: Decimal
    currency: str
    payment_method: str
    current_status: str
    source: str = "LIVE_SYSTEM_DATABASE"
    events: list[AuditEvent]

File: frontend/src/types/audit.ts
export type AuditDetailValue = string | number | boolean | null | AuditDetailValue[] | { [key:string]: AuditDetailValue };
export interface AuditEvent { event_type:string; title:string; status:string|null; timestamp:string|null; details:Record<string,AuditDetailValue> }
export interface PaymentAudit { payment_id:number; amount:string; currency:string; payment_method:string; current_status:string; source:string; events:AuditEvent[] }
